src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/Event.scala

```diff
@@ -242,12 +242,202 @@ case class Event(
+      validateStr("refr_urlpath", e.refr_urlpath) ++
+      validateStr("refr_urlquery", e.refr_urlquery) ++
+      validateStr("refr_urlfragment", e.refr_urlfragment) ++
+      validateStr("refr_medium", e.refr_medium) ++
+      validateStr("refr_source", e.refr_source) ++
+      validateStr("refr_term", e.refr_term) ++
+      validateStr("mkt_medium", e.mkt_medium) ++
+      validateStr("mkt_source", e.mkt_source) ++
+      validateStr("mkt_term", e.mkt_term) ++
+      validateStr("mkt_content", e.mkt_content) ++
+      validateStr("mkt_campaign", e.mkt_campaign) ++
+      validateStr("se_category", e.se_category) ++
+      validateStr("se_action", e.se_action) ++
+      validateStr("se_label", e.se_label) ++
+      validateStr("se_property", e.se_property) ++
+      validateStr("tr_orderid", e.tr_orderid) ++
+      validateStr("tr_affiliation", e.tr_affiliation) ++
+      validateStr("tr_city", e.tr_city) ++
+      validateStr("tr_state", e.tr_state) ++
+      validateStr("tr_country", e.tr_country) ++
+      validateStr("ti_orderid", e.ti_orderid) ++
+      validateStr("ti_sku", e.ti_sku) ++
+      validateStr("ti_name", e.ti_name) ++
+      validateStr("ti_category", e.ti_category) ++
+      validateStr("useragent", e.useragent) ++
+      validateStr("br_name", e.br_name) ++
+      validateStr("br_family", e.br_family) ++
+      validateStr("br_version", e.br_version) ++
+      validateStr("br_type", e.br_type) ++
+      validateStr("br_renderengine", e.br_renderengine) ++
+      validateStr("br_lang", e.br_lang) ++
+      validateStr("br_colordepth", e.br_colordepth) ++
+      validateStr("os_name", e.os_name) ++
+      validateStr("os_family", e.os_family) ++
+      validateStr("os_manufacturer", e.os_manufacturer) ++
+      validateStr("os_timezone", e.os_timezone) ++
+      validateStr("dvce_type", e.dvce_type) ++
+      validateStr("doc_charset", e.doc_charset) ++
+      validateStr("tr_currency", e.tr_currency) ++
+      validateStr("ti_currency", e.ti_currency) ++
+      validateStr("base_currency", e.base_currency) ++
+      validateStr("geo_timezone", e.geo_timezone) ++
+      validateStr("mkt_clickid", e.mkt_clickid) ++
+      validateStr("mkt_network", e.mkt_network) ++
+      validateStr("etl_tags", e.etl_tags) ++
+      validateStr("refr_domain_userid", e.refr_domain_userid) ++
+      validateStr("domain_sessionid", e.domain_sessionid) ++
+      validateStr("event_vendor", e.event_vendor) ++
+      validateStr("event_name", e.event_name) ++
+      validateStr("event_format", e.event_format) ++
+      validateStr("event_version", e.event_version) ++
+      validateStr("event_fingerprint", e.event_fingerprint)
+
   /**
    * Automatically derived Circe encoder
    */
   implicit val jsonEncoder: Encoder.AsObject[Event] = deriveEncoder[Event]
 
-  implicit def eventDecoder: Decoder[Event] = deriveDecoder[Event]
+  implicit def eventDecoder: Decoder[Event] = deriveDecoder[Event].ensure(validator)
```

Comment:
In [the origin issue](https://github.com/snowplow/snowplow-scala-analytics-sdk/issues/115#issuecomment-975642378) Paul requested that we implemented it so that validation is optional.  Will it be optional if you add it onto this encoder?